### PR TITLE
[Spec] Added spec of `code`, `resource_account` & `staking_proxy` module

### PR DESCRIPTION
**Pull request by xudon9**
### Description
This is a part of spec work for Aptos Framework from MoveBit. This PR added specs for following modules:
-  `code` 
- `resource_account`
- `staking_proxy` 
### Test Plan
Using command `aptos prove -f FILE` for each files

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/code.spec.move [hunk 1]
```diff
@@ -12,4 +13,35 @@ spec aptos_framework::code {
...
+        // TODO: Can't verify `check_upgradability` in while loop. 
+        pragma verify = false;
+    }
+
+    spec publish_package_txn {
+        // TODO: Calls `publish_package`.`
+        pragma verify = false;
+    }
+
+    spec check_upgradability(old_pack: &PackageMetadata, new_pack: &PackageMetadata, new_modules: &vector<String>) {
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/code.spec.move:35]: Could you put the TODO marks and the reason why?

File: aptos-move/framework/aptos-framework/sources/code.spec.move [hunk 2]
```diff
@@ -12,4 +13,35 @@ spec aptos_framework::code {
...
+        pragma verify = false;
+    }
+
+    spec check_dependencies(publish_address: address, pack: &PackageMetadata): vector<AllowedDep> {
+        pragma verify = false;
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/code.spec.move:40]: TODO?

File: aptos-move/framework/aptos-framework/sources/code.spec.move [hunk 3]
```diff
@@ -12,4 +13,35 @@ spec aptos_framework::code {
...
+    }
+
+    spec check_coexistence(old_pack: &PackageMetadata, new_modules: &vector<String>) {
+        pragma verify = false;
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/code.spec.move:44]: TODO?

**Review by junkil-park — APPROVED** [review]
TODOs should be taken care of in the following PRs.

**Review by rahxephon89 — APPROVED** [review]

**Pull request merged**